Pass TokenSwap state into the constructor to simplify function calls

rawfalafel opened the pull request:
Addresses #406

joncinque commented on token-swap/js/client/token-swap.js:
```diff
@@ -328,23 +380,16 @@ export class TokenSwap {
 
   /**
-   * Swap the tokens in the pool
+   * Swap token A for token B
```
The next step of token-swap is to expand to pools, so the function will need to take in the accounts of the swap no matter what -- meaning this can keep the same signature as before, minus the `authority` and `tokenProgramId`.  Going along with that, you can get rid of the second function `swapInverse`

rawfalafel replied:
Gotcha. Since devs can fetch the token mint accounts with `getSwapInfo`, requiring them to pass in the mint accounts here seems more reasonable.

joncinque commented on token-swap/js/client/token-swap.js:
```diff
@@ -328,23 +380,16 @@ export class TokenSwap {
-   * @param tokenProgramId Token program id
-   * @param amount Amount to transfer from source account
+   * @param amountIn Amount to transfer from source account
+   * @param minimumAmountOut Minimum amount to send
```
This param should be "Minimum amount to receive"

rawfalafel replied:
Isn't `amountIn` the amount the program's receiving, and `minimumAmountOut` the amount the program's sending?

joncinque replied:
Ah, my mistake, I misunderstood.  Yes, `minimumAmountOut` is how much the program sends out, but I thought the comment implied how much the user was sending in.  Since this `token-swap` js package is user-facing, the parameters should be framed from the point of view of the user, and not the program.  The user is sending in `amountIn`, and receiving `minimumAmountOut`.  Does that make sense?

rawfalafel replied:
yep!

joncinque reviewed (commented):
Looks good for the most part, just the one part about the `swap` function which needs to be updated.  Otherwise, it looks like this failed the lint test, so be sure to run `npm run lint:fix` as well.

joncinque commented:
And also, could you update the signatures in `module.flow.js` and `module.d.ts`?

rawfalafel commented on Cargo.lock:
```diff
@@ -561,26 +561,26 @@ dependencies = [
 [[package]]
 name = "curve25519-dalek"
 version = "2.1.0"
-source = "git+https://github.com/garious/curve25519-dalek?rev=60efef3553d6bf3d7f3b09b5f97acd54d72529ff#60efef3553d6bf3d7f3b09b5f97acd54d72529ff"
```
I couldn't figure out what was causing this file to automatically change for me. Any idea?

rawfalafel commented:
@joncinque this PR now changes the program's state to include `tokenProgramId`. In the process, I added a `loadTokenSwap` static method and removed the `TokenSwapInfo` struct entirely. Let me know if it makes sense to you!

joncinque commented on token-swap/js/client/token-swap.js:
```diff
@@ -342,54 +359,21 @@ export class TokenSwap {
+   * @param userSource User's source token account
+   * @param poolSource Pool's source token account
+   * @param userSource User's destination token account
+   * @param poolSource Pool's destination token account
```
Comments here reference the wrong variables `userSource` -> `userDestination` and `poolSource` -> `poolDestination`

joncinque commented on token-swap/js/client/token-swap.js:
```diff
@@ -342,54 +359,21 @@ export class TokenSwap {
+    userSource: PublicKey,
+    poolSource: PublicKey,
+    userDestination: PublicKey,
+    poolDestination: PublicKey,
```
Could you invert these two parameters to match `swapInstruction`?

joncinque commented on token-swap/js/client/token-swap.js:
```diff
@@ -452,10 +401,10 @@ export class TokenSwap {
+    userSource: PublicKey,
+    poolSource: PublicKey,
+    userDestination: PublicKey,
+    poolDestination: PublicKey,
```
same here

joncinque commented on token-swap/js/client/token-swap.js:
```diff
@@ -480,10 +429,10 @@ export class TokenSwap {
+      {pubkey: userSource, isSigner: false, isWritable: true},
+      {pubkey: poolSource, isSigner: false, isWritable: true},
+      {pubkey: userDestination, isSigner: false, isWritable: true},
+      {pubkey: poolDestination, isSigner: false, isWritable: true},
```
last here

joncinque commented on token-swap/js/module.d.ts:
```diff
@@ -16,23 +16,20 @@ declare module '@solana/spl-token-swap' {
     static fromBuffer(buffer: Buffer): Numberu64;
   }
 
-  export type TokenSwapInfo = {
```
nice job removing this!

joncinque commented on token-swap/js/module.d.ts:
```diff
@@ -70,26 +74,22 @@ declare module '@solana/spl-token-swap' {
+      userSource: PublicKey,
+      poolSource: PublicKey,
+      userDestination: PublicKey,
+      poolDestination: PublicKey,
```
these should also be put back as they were

joncinque commented on token-swap/js/module.flow.js:
```diff
@@ -66,26 +70,22 @@ declare module '@solana/spl-token-swap' {
+      userSource: PublicKey,
+      poolSource: PublicKey,
+      userDestination: PublicKey,
+      poolDestination: PublicKey,
```
same here

joncinque commented on token-swap/js/module.flow.js:
```diff
@@ -66,26 +70,22 @@ declare module '@solana/spl-token-swap' {
+      userSource: PublicKey,
+      poolSource: PublicKey,
+      userDestination: PublicKey,
+      poolDestination: PublicKey,
```
same here

joncinque reviewed (commented):
just a few more small things, then this should be good to go!

rawfalafel commented:
good catch on the parameters names. this should be ready for another look.

joncinque commented:
Looks great!  It looks like CI didn't get triggered, so I'm going to close / re-open this to see if that gets it going

joncinque commented:
All looks good -- thanks for the contribution
